src/libretro/libretro.cpp

```diff
@@ -325,4 +324,78 @@ void Platform::WriteFirmware(const Firmware& firmware, u32 writeoffset, u32 writ
     ZoneScopedN(TracyFunction);
 
     MelonDsDs::Core.WriteFirmware(firmware, writeoffset, writelen);
-}
\ No newline at end of file
+}
+
+extern "C" void MelonDsDs::MpStarted(uint16_t client_id, retro_netpacket_send_t send_fn, retro_netpacket_poll_receive_t poll_receive_fn) {
+    MelonDsDs::Core.MpStarted(send_fn, poll_receive_fn);
+}
+
+extern "C" void MelonDsDs::MpReceived(const void* buf, size_t len, uint16_t client_id) {
+    MelonDsDs::Core.MpPacketReceived(buf, len);
+}
+
+extern "C" void MelonDsDs::MpStopped() {
+    MelonDsDs::Core.MpStopped();
+}
+
+int DeconstructPacket(u8 *data, u64 *timestamp, std::optional<MelonDsDs::Packet> o_p) {
+    if (!o_p.has_value()) {
+        return 0;
+    }
+    MelonDsDs::Packet p = o_p.value();
+    memcpy(data, p.Data(), p.Length());
+    *timestamp = p.Timestamp();
+    return p.Length();
+}
+
+int Platform::MP_SendPacket(u8* data, int len, u64 timestamp, void*) {
+    return MelonDsDs::Core.MpSendPacket(MelonDsDs::Packet(data, len, timestamp, 0, false)) ? len : 0;
+}
+
+int Platform::MP_RecvPacket(u8* data, u64* timestamp, void*) {
+    std::optional<MelonDsDs::Packet> o_p = MelonDsDs::Core.MpNextPacket();
+    return DeconstructPacket(data, timestamp, o_p);
+}
+
+int Platform::MP_SendCmd(u8* data, int len, u64 timestamp, void*) {
+    return MelonDsDs::Core.MpSendPacket(MelonDsDs::Packet(data, len, timestamp, 0, false)) ? len : 0;
+}
+
+int Platform::MP_SendReply(u8 *data, int len, u64 timestamp, u16 aid, void*) {
+    retro_assert(aid < 16);
```

Comment:
Because melonDS seems to only send aid s less than 16, otherwise sending a 16-bit wide aid mask in RecvReplies would make no sense, and `ret |= 1<<aid` in RecvReplies would also make no sense, and I pretty much copied that line from [melonDS itself](https://github.com/melonDS-emu/melonDS/blob/817b409ec893fb0b2b745ee18feced08706419de/src/net/LAN.cpp#L1074).
That assert is documenting what seems to be a characteristic of the protocol which I did not find documented anywhere else.
Would you prefer if I left a code comment explaining this?

Reply:
Yes, please. If that assumption is ever violated, it would be helpful to know why it was made in the first place.